dask/diagnostics/profile_visualize.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import division, absolute_import
+
+from itertools import cycle
+from operator import itemgetter
+
+from toolz import unique, groupby, valmap
+import bokeh.plotting as bp
+from bokeh.palettes import brewer
+from bokeh.models import HoverTool
+
+from ..dot import name
+
+
+def get_colors(palette, funcs):
+    """Get a dict mapping funcs to colors from palette.
+
+    Parameters
+    ----------
+    palette : string
+        Name of the palette. Must be a key in bokeh.palettes.brewer
+    funcs : iterable
+        Iterable of function names
+    """
+    unique_funcs = list(sorted(unique(funcs)))
+    n_funcs = len(unique_funcs)
+    palette_lookup = brewer[palette]
+    keys = list(palette_lookup.keys())
+    low, high = min(keys), max(keys)
+    if n_funcs > high:
+        colors = cycle(palette_lookup[high])
+    elif n_funcs < low:
+        colors = palette_lookup[low]
+    else:
+        colors = palette_lookup[n_funcs]
+    color_lookup = dict(zip(unique_funcs, colors))
+    return [color_lookup[n] for n in funcs]
+
+
+def visualize(results, palette='GnBu', file_path="profile.html",
+              show=True, **kwargs):
+    """Visualize the results of profiling in a bokeh plot.
+
+    Parameters
+    ----------
+    results : sequence
+        Output of profiler.results().
+    palette : string, optional
+        Name of the bokeh palette to use, must be key in bokeh.palettes.brewer.
+    file_path : string, optional
+        Name of the plot output file.
+    show : boolean, optional
+        If True (default), the plot is opened in a browser.
+    **kwargs
+        Other keyword arguments, passed to bokeh.figure. These will override
+        all defaults set by visualize.
+
+    Returns
+    -------
+    The completed bokeh plot object.
+    """
+
+    bp.output_file(file_path)
+    keys, tasks, starts, ends, ids = zip(*results)
+
+    id_group = groupby(itemgetter(4), results)
+    diff = lambda v: v[3] - v[2]
+    timings = valmap(lambda val: sum(map(diff, val)), id_group)
+    id_lk = {t[0]: n for (n, t) in enumerate(sorted(timings.items(),
+                                             key=itemgetter(1), reverse=True))}
+
+    left = min(starts)
+    right = max(ends)
+
+    defaults = dict(title="Profile Results",
+                    tools="hover,save,reset,resize,xwheel_zoom,xpan",
+                    plot_width=800, plot_height=400)
```

Comment:
It would be nice to accept width and height as kwargs.

Perhaps we should test that kwargs pass through.  Tests should be possible by setting `show=False` and then inspecting the resulting figure object.